src/simple_tracker_detection_provider/simple_tracker_detection_provider/blob_detector.py

```diff
@@ -0,0 +1,106 @@
+# Original work Copyright (c) 2022 Sky360
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+import cv2
+from simple_tracker_shared.utils import kp_to_bbox
+import pysky360 as sky360
+
+class BlobDetector():
+
+    # Static factory select method to determine what masking implementation to use
+    @staticmethod
+    def Select(settings):
+
+        detector_type = settings['blob_detector_type']
+
+        if detector_type == 'simple':
+            return BlobDetector.Simple(settings)
+
+        if detector_type == 'sky360':
+            return BlobDetector.Sky360(settings)
+
+    @staticmethod
+    def Simple(settings):
+        return SimpleBlobDetector(settings)
+
+    @staticmethod
+    def Sky360(settings):
+        return Sky360BlobDetector(settings)
+
+
+    def __init__(self, settings):
+        self.settings = settings
+        self.sensitivity = settings['tracker_detection_sensitivity']
+
+    def detect(self, frame):
+        pass
+
+class SimpleBlobDetector(BlobDetector):
+
+    def __init__(self, settings):
+        super().__init__(settings)
+
+        params = cv2.SimpleBlobDetector_Params()
+
+        params.minRepeatability = 2
+        params.minDistBetweenBlobs = 32
+        params.minThreshold = 3
+        params.filterByArea = 1
+        params.filterByColor = 0
+        # params.blobColor=255
+
+        if self.sensitivity == 1:  # Detects small, medium and large objects
```

Comment:
Should "sensitivity" be an enumeration for clarity? You don't want callers to have to understand this mapping.

Reply:
This is a very fair point, will look into this in the coming week